### PR TITLE
Interrupt sync

### PR DESCRIPTION
**Pull request by zancas**
Let's get this implemented next!

### CODE REVIEW

File: lib/src/lightclient.rs [hunk 1]
```diff
@@ -140,6 +141,7 @@ impl LightClient {
         &self,
         addrs: Vec<(&str, u64, Option<String>)>,
     ) -> Result<String, String> {
+        let transaction_submission_height = self.latest_served_blockheight().await;
```
Comment (AloeareV) [inline lib/src/lightclient.rs:144]: Submission target should be 1 above latest served blockheight, we're aiming for the next block, not the one already mined.

File: lib/src/lightclient.rs [hunk 2]
```diff
@@ -1517,6 +1540,7 @@ impl LightClient {
     }
 
     pub async fn do_shield(&self, address: Option<String>) -> Result<String, String> {
+        let transaction_submission_height = self.latest_served_blockheight().await;
```
Comment (AloeareV) [inline lib/src/lightclient.rs:1543]: Same as above

File: lib/src/lightclient.rs [hunk 3]
```diff
@@ -1559,8 +1584,20 @@ impl LightClient {
         result.map(|(transaction_id, _)| transaction_id)
     }
 
+    async fn latest_served_blockheight(&self) -> BlockHeight {
+        let latest_height = GrpcConnector::get_latest_block(self.config.get_server_uri())
+            .await
+            .unwrap()
+            .height;
+        BlockHeight::from_u32((latest_height - self.config.reorg_buffer_offset as u64) as u32)
```
Comment (AloeareV) [inline lib/src/lightclient.rs:1592]: This function's name mislead me into think this function returns the latest server blockheight, not the latest served height minus our intended anchor offset.

Comment (AloeareV) [reply lib/src/lightclient.rs:1592]: I think the function is named what it wants to do, and this `-` bit should be removed.

File: lib/src/wallet.rs [hunk 4]
```diff
@@ -632,7 +632,7 @@ impl LightWallet {
             .unwrap_or_default()
     }
 
-    async fn get_target_height(&self) -> Option<u32> {
+    async fn get_wallet_latest_height(&self) -> Option<u32> {
```
Comment (AloeareV) [inline lib/src/wallet.rs:635]: This function returns 1 + latest height, so calling it `wallet_latest_height` is incorrect.

File: lib/src/wallet.rs [hunk 5]
```diff
@@ -1186,19 +1190,17 @@ impl LightWallet {
...
         println!("Selected notes worth {}", u64::from(selected_value));
 
-        let orchard_anchor = self
-            .get_orchard_anchor(&orchard_notes, target_height)
-            .await?;
+        let orchard_anchor = self.get_orchard_anchor(&orchard_notes, last_height).await?;
         let mut builder = Builder::with_orchard_anchor(
             self.transaction_context.config.chain,
-            target_height,
+            submission_height,
```
Comment (AloeareV) [inline lib/src/wallet.rs:1203]: Right now, this is `reorg_buffer_offset + 1` less than what the submission height should be, which is one above the highest block known to the server.

**Review by AloeareV — CHANGES_REQUESTED** [review]

File: lib/src/wallet.rs (hunk 4 above)
Comment (zancas) [reply lib/src/wallet.rs:635]: Wait...   why does this function return 1 + last_synced_height?

Comment (zancas) [reply lib/src/wallet.rs]: Yes.

**Review by AloeareV — APPROVED** [review]
Great!

**Pull request merged**